pkg/resources/object_renaming_acceptance_test.go

```diff
@@ -0,0 +1,1366 @@
+- are not in config - database renamed externally - referencing new database name (fails because tries to create table that already exists in the renamed database)
+
+- are not in config - schema renamed externally - referencing old schema name (fails because tries to create table on non-existing schema)
+- are not in config - schema renamed externally - referencing new schema name (fails because tries to create table that already exists in the renamed schema)
+
+# The list of test cases that were not added:
+- (Deep hierarchy) More test cases with varying dependencies between resources
+- (Deep hierarchy) Add test cases where old database is referenced to see if hierarchy recreation is possible
+- (Deep hierarchy) More test cases could be added when database and schema are renamed at the same time
+- (Deep hierarchy) More test cases could be added when either database or schema are in the config
+*/
+
+type DependencyType string
+
+const (
+	ImplicitDependency  DependencyType = "implicit"
+	DependsOnDependency DependencyType = "depends_on"
+	NoDependency        DependencyType = "no_dependency"
+)
+
+func TestAcc_ShallowHierarchy_IsInConfig_RenamedInternally_WithImplicitDependency(t *testing.T) {
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableObjectRenamingTest)
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableAcceptance)
+	acc.TestAccPreCheck(t)
+
+	databaseId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	newDatabaseId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	schemaName := acc.TestClient().Ids.Alpha()
+
+	databaseConfigModel := model.Database("test", databaseId.Name())
+	databaseConfigModelWithNewId := model.Database("test", newDatabaseId.Name())
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.Schema),
+		Steps: []resource.TestStep{
+			{
+				Config: config.FromModel(t, databaseConfigModel) + configSchemaWithDatabaseReference(databaseConfigModel.ResourceReference(), schemaName),
+			},
+			{
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction("snowflake_database.test", plancheck.ResourceActionUpdate),
+						plancheck.ExpectResourceAction("snowflake_schema.test", plancheck.ResourceActionDestroyBeforeCreate),
+					},
+				},
+				Config: config.FromModel(t, databaseConfigModelWithNewId) + configSchemaWithDatabaseReference(databaseConfigModelWithNewId.ResourceReference(), schemaName),
+			},
+		},
+	})
+}
+
+func TestAcc_ShallowHierarchy_IsInConfig_RenamedInternally_WithoutDependency_AfterRenameSchemaReferencingOldDatabaseName(t *testing.T) {
+	// Error happens during schema's Read operation and then Delete operation (schema cannot be removed).
```

Comment:
We can, but the overall test will fail anyway. It's currently failing because the terraform is trying to remove schema on a non-existing database, so dropping it by hand wouldn't do anything.